src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -386,7 +381,16 @@ public virtual void Challenge(IChallengeContext context)
         /// <returns></returns>
         protected virtual Task ApplyResponseChallengeAsync()
         {
-            ApplyResponseChallenge();
+            // If authenticate was called and the the status is still 401, authZ failed so set 403 and stop
+            // REVIEW: Does this need to ensure that there's the 401 is challenge for this auth type?
+            if (ShouldConvertChallengeToForbidden())
```

Comment:
Okay I'll move the logic and also add a test verifying that Cookie/Bearer don't turn 401's into 403s if they weren't asked to produce a ticket